pkgs/applications/audio/littlegptracker/default.nix

```diff
@@ -0,0 +1,81 @@
+{ stdenv
+, fetchFromGitHub
+, SDL
+, jack2
+, Foundation
+}:
+
+stdenv.mkDerivation rec {
+  pname = "littlegptracker";
+  version = "unstable-2019-04-14";
+
+  src = fetchFromGitHub {
+    owner = "Mdashdotdashn";
+    repo = "littlegptracker";
+    rev = "0ed729b46739e3df5e111c6fa4d548fde2d3b891";
+    sha256 = "1pc6lg2qp6xh7ahs5d5pb63ms4h2dz7ryp3c7mci4g37gbwbsj5b";
+  };
+
+  buildInputs = [
+    SDL
+    jack2
+  ] ++ stdenv.lib.optional stdenv.isDarwin Foundation;
+
+  postPatch = ''
+    # Remove coherency checks >:)
+    # As far as I can see, they are just to make sure that the types can fit
+    # in the word length, but this check was written when 64 bits weren't a
+    # possibility and didn't take that into account,
+    # so although the types do fit, the checks fail.
+    # Indeed, the program seems to work well on 64 bits.
+    # More info here:
+    # https://github.com/Mdashdotdashn/LittleGPTracker/issues/4
+    sed -i '126,147d' sources/Externals/Soundfont/DATATYPE.H
+
+    # Set the initial directory to the current directory, otherwise the user has
+    # to navigate from the nix store, which makes the program crash due to its
+    # size
+    sed -i 's@"bin:\.\."@"."@' sources/Adapters/DEB/System/DEBSystem.cpp
+  '';
+
+  preBuild = "cd projects";
+
+  makeFlags =
+    stdenv.lib.optionals stdenv.isLinux  [ "PLATFORM=DEB" ] ++
+    stdenv.lib.optionals stdenv.isDarwin [
+      "PLATFORM=OSX"
+      "CXX=${stdenv.cc.targetPrefix}cc" # use clang, not g++
```

Comment:
Does Clang on Linux work without this? If not, making this a platform-independent makeFlag might make sense.

Reply:
I should clarify the comment: it makes it use clang _on darwin_, where cc=clang

Reply:
Oh wait, I think I misunderstood your comment, sorry.

Hmm yes, I think this could be more generic. Let me try...
(also I wonder why CXX isn't set by default)

Reply:
> (also I wonder why CXX isn't set by default)

The build process re-exports `CC` and `CXX`, amongst afew other variables. I believe `:=` as it is used there drops the previously assigned value, while `?=` for example would only set the value if it hasn't already been set? See [projects/base_rules](https://github.com/Mdashdotdashn/LittleGPTracker/blob/7088032800b0ba30e80e363102b23c07d79006cc/projects/base_rules#L12-L16).

Reply:
```sh
substituteInPlace projects/base_rules --replace ":=" "?="
```
might indeed be enough to resolve this whole thing? I tried using `clangStdenv` on Linux with this change and it correctly started using `clang++` (though the build didn't get far, but Clang might just not be supported for the Linux build). For giggles I tried cross-compiling for mingw32 and that correctly picked up `i686-w64-mingw32-g++` as well (also fails although technically supported, afew things about the Windows part seem unmaintained so prolly not worth the effort).